### DISCUSSION TITLE
`Ctrl + u` for discarding line in `Picker`?

### DISCUSSION THREAD

**Discussion by grantshandy** [helix-editor/helix#10429] [General]
<kbd>Ctrl</kbd> + <kbd>u</kbd> is a common keyboard shortcut for clearing the current input throughout shell environments and command lines. Helix uses <kbd>Ctrl</kbd> + <kbd>u</kbd> to clear the input for `:` commands, could it be added to the symbol viewer/global search/file picker/LSP references?

I keep reaching for the keyboard shortcut when I want to reset my query, instead I have to hold the backspace key. Does this sound like something that should be added? If it seems like a feature that would be added I can try to make a PR.

**Comment by grantshandy** [comment]
Ah, I see. <kbd>Ctrl</kbd> + <kbd>u</kbd>/<kbd>d</kbd> is for Page Up/Down in the list as added in #1612.

After more research, I've found that <kbd>Ctrl</kbd> + <kbd>u</kbd> is commonly called "unix-line-discard". Now my main question is why shouldn't this common (arguably canonical) shortcut not be used in `Picker` when it is already used in the command entry?

**Comment by d2weber** [comment]
Duplicate of #1593

**Comment by gabydd** [comment]
This is different then that issue it happens because the picker and the prompt have a conflicting key in common so ``Ctrl-u`` gets consumed by the picker before it reaches the prompt

**Comment by d2weber** [comment]
> This is different then that issue it happens because the picker and the prompt have a conflicting key in common so `Ctrl-u` gets consumed by the picker before it reaches the prompt

I see, you are right

**Comment by david-amirault** [comment]
+1. Unix-line-discard is **much** more useful than page up/page down in picker. If you need to scroll pages in picker, then you are likely using it wrong, and your fuzzy-matching text should be more specific. Unix-line-discard works in all other helix text-entry modes, so it is quite unintuitive that it does not work in picker.

Even once we gain the ability to remap keys in picker, I feel strongly that unix-line-discard should be the default for Ctrl-u in picker

**Reply by robinsonrc** [reply]
Agreed, I was surprised when this wasn’t the behaviour and didn’t match the command prompt or shell in general
